### PR TITLE
update ARG example for v2

### PR DESCRIPTION
**Pull request by claresinger**
Add lines to ARG plots from CloudMicrophysics.jl activation parameterization (just Fig 1 for now)
Looks good: [ARG_fig1.pdf](https://github.com/atmos-cloud-sim-uj/PySDM-examples/files/8544387/ARG_fig1.pdf)

- [x] Generate data from CloudMicrophysics.jl for Figs 2-5; with help from @trontrytel 
- [x] Add these lines to Figs 2-5; task for @claresinger 

Updated to include `B` or `κ` formulation for all figures. 
- [x] Should choose just one to show in the JOSS v2 paper.

Closes https://github.com/atmos-cloud-sim-uj/PySDM/issues/813
- [x] Add Colab install header

### CODE REVIEW

**Comment by claresinger** [comment]
For now I'm just using the `B` formulation of the parameterization in CloudMicrophysics.jl. 
TODO: remove the kappa line from Fig 1 to be consistent.

**Comment by claresinger** [comment]
Ready to merge with approval from @trontrytel and @slayoo :)

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 1]
```diff
@@ -43,15 +43,1680 @@
...
     "    Fig5b_w_obs, Fig5b_AF_obs, Fig5b_w_param, Fig5b_AF_param\n",
     ")\n",
+    "from PySDM_examples.Abdul_Razzak_Ghan_2000.data_from_CloudMicrophysics_ARG import (\n",
+    "    Fig1_N2_param_jl, Fig1_AF_param_jl_B,\n",
+    "    Fig2_N2_param_jl, Fig2a_AF_param_jl_B, Fig2b_AF_param_jl_B,\n",
+    "    Fig3_sol2_param_jl, Fig3a_AF_param_jl_B, Fig3b_AF_param_jl_B,\n",
+    "    Fig4_rad2_param_jl, Fig4a_AF_param_jl_B, Fig4b_AF_param_jl_B,\n",
+    "    Fig5_w_param_jl, Fig5a_AF_param_jl_B, Fig5b_AF_param_jl_B,\n",
+    ")\n",
+    "# from PySDM_examples.Abdul_Razzak_Ghan_2000.data_from_CloudMicrophysics_ARG import (\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:53]: @claresinger to make it maintainable, I'd suggest either removing the commented out code or uncommenting it

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 2]
```diff
@@ -72,6 +1737,9 @@
     "ax.plot(Fig1_N2_obs, Fig1_AF_obs, \"ko\", label=\"ARG 2000 data\")\n",
     "ax.plot(Fig1_N2_param, Fig1_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
     "\n",
+    "ax.plot(Fig1_N2_param_jl, Fig1_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# ax.plot(Fig1_N2_param_jl, Fig1_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:1741]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 3]
```diff
@@ -107,8 +3702,12 @@
     "fig, axes = pyplot.subplots(2, 1, sharex=True, figsize=(6,6))\n",
     "axes[0].plot(Fig2a_N2_obs, Fig2a_AF_obs, \"ko\", label=\"ARG 2000 data\")\n",
     "axes[0].plot(Fig2a_N2_param, Fig2a_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[0].plot(Fig2_N2_param_jl, Fig2a_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[0].plot(Fig2_N2_param_jl, Fig2a_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:3706]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 4]
```diff
@@ -107,8 +3702,12 @@
...
     "axes[1].plot(Fig2b_N2_obs, Fig2b_AF_obs, \"ko\")\n",
     "axes[1].plot(Fig2b_N2_param, Fig2b_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[1].plot(Fig2_N2_param_jl, Fig2b_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[1].plot(Fig2_N2_param_jl, Fig2b_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:3710]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 5]
```diff
@@ -146,8 +5551,12 @@
     "fig, axes = pyplot.subplots(2, 1, sharex=True, figsize=(6,6))\n",
     "axes[0].plot(Fig3a_sol2_obs, Fig3a_AF_obs, \"ko\", label=\"ARG 2000 data\")\n",
     "axes[0].plot(Fig3a_sol2_param, Fig3a_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[0].plot(Fig3_sol2_param_jl, Fig3a_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[0].plot(Fig3_sol2_param_jl, Fig3a_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:5555]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 6]
```diff
@@ -146,8 +5551,12 @@
...
     "axes[1].plot(Fig3b_sol2_obs, Fig3b_AF_obs, \"ko\")\n",
     "axes[1].plot(Fig3b_sol2_param, Fig3b_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[1].plot(Fig3_sol2_param_jl, Fig3b_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[1].plot(Fig3_sol2_param_jl, Fig3b_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:5559]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 7]
```diff
@@ -185,8 +7600,12 @@
     "fig, axes = pyplot.subplots(2, 1, sharex=True, figsize=(6,6))\n",
     "axes[0].plot(np.asarray(Fig4a_rad2_obs)*1e3, Fig4a_AF_obs, \"ko\", label=\"ARG 2000 data\")\n",
     "axes[0].plot(np.asarray(Fig4a_rad2_param)*1e3, Fig4a_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[0].plot(np.asarray(Fig4_rad2_param_jl)*1e3, Fig4a_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[0].plot(np.asarray(Fig4_rad2_param_jl)*1e3, Fig4a_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:7604]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 8]
```diff
@@ -185,8 +7600,12 @@
...
     "axes[1].plot(np.asarray(Fig4b_rad2_obs)*1e3, Fig4b_AF_obs, \"ko\")\n",
     "axes[1].plot(np.asarray(Fig4b_rad2_param)*1e3, Fig4b_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[1].plot(np.asarray(Fig4_rad2_param_jl)*1e3, Fig4b_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[1].plot(np.asarray(Fig4_rad2_param_jl)*1e3, Fig4b_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:7608]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 9]
```diff
@@ -226,8 +7645,12 @@
     "fig, axes = pyplot.subplots(2, 1, sharex=True, figsize=(6,6))\n",
     "axes[0].plot(Fig5a_w_obs, Fig5a_AF_obs, \"ko\", label=\"ARG 2000 data\")\n",
     "axes[0].plot(Fig5a_w_param, Fig5a_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[0].plot(Fig5_w_param_jl, Fig5a_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[0].plot(Fig5_w_param_jl, Fig5a_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:7649]: ditto

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 10]
```diff
@@ -226,8 +7645,12 @@
...
     "axes[1].plot(Fig5b_w_obs, Fig5b_AF_obs, \"ko\")\n",
     "axes[1].plot(Fig5b_w_param, Fig5b_AF_param, \"k-\", label=\"ARG 2000 param\")\n",
+    "axes[1].plot(Fig5_w_param_jl, Fig5b_AF_param_jl_B, \"k--\", label=\"CloudMicrophysics.jl param (B)\")\n",
+    "# axes[1].plot(Fig5_w_param_jl, Fig5b_AF_param_jl_k, \"k-.\", label=r\"CloudMicrophysics.jl param ($\\kappa$)\")\n",
```
Comment (slayoo) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:7653]: and the last one :)

**Comment by trontrytel** [comment]
I confirm that the notebook works for me locally. 

Maybe a tiny two details with regards to the plots:

- Would it be nicer to plot the PySDM results with different marker shape for Scrit vs Vcrit option? This way, for cases when the results overlap each other, one could still see one marker under the other marker?

- For Fig 4, would it make sense to make the y-axis go a little bit over 1?

**Comment by slayoo** [comment]
@claresinger let's also add pacage installation header cell for Colab, i.e.:
```python
import sys
if 'google.colab' in sys.modules:
    !pip --quiet install atmos-cloud-sim-uj-utils
    from atmos_cloud_sim_uj_utils import pip_install_on_colab
    pip_install_on_colab('PySDM-examples')
```
This will close #813

**Comment by slayoo** [comment]
:tada:

**Comment by trontrytel** [comment]
@claresinger, @slayoo the PySDM examples is becoming a really impressive project. Really cool thing to witness

**Comment by trontrytel** [comment]
If you ever wanted to make a case for open-source software for science, here you have one ;)

**Pull request merged**